pkgs/build-support/fetchgit/tests.nix

```diff
@@ -113,6 +127,18 @@
     '';
   };
 
+  git-describe-without-dot-git = testers.invalidateFetcherByDrvHash fetchgit {
+    name = "fetchgit-git-describe-without-dot-git-test";
+    url = "https://github.com/NixOS/nix";
+    rev = "9d9dbe6ed05854e03811c361a3380e09183f4f4a";
+    fetchTags = true;
+    leaveDotGit = false;
+    hash = "sha256-y7l+46lVP2pzJwGON5qEV0EoxWofRoWAym5q9VXvpc8=";
+    postCheckout = ''
+      { git -C "$out" describe --tags --always 2>&1 || echo "git describe failed"; } | tee "$out/describe-output.txt"
+    '';
+  };
+
```

Comment:
Ah, now I see where the concerns about `git describe` are coming from. I definitely think it's possible to check if the `.git` repository is still present and useful with a less fragile command.

Perhaps adding `rec` to the function args and changing `postCheckout` to something like the below?

```
expected_rev=${lib.escapeShellArg rev}
git -C "$out" rev-parse HEAD | tee "$out"/head.txt
head_rev="$(<"$out"/head.txt)"
if [[ "$head_rev" != "$expected_rev" ]]; then
    echo 'unexpected revision at HEAD' >&2
    echo "expected $expected_rev" >&2
    echo "found $head_rev" >&2
    exit 1
fi
```